modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Apply.kt

```diff
@@ -44,13 +44,28 @@ interface Apply<F> : Functor<F> {
    */
   fun <A, B> Kind<F, A>.lazyAp(ff: () -> Kind<F, (A) -> B>): Kind<F, B> = ap(ff())
 
+  @Deprecated(
+    "map is being renamed to mapN",
+    ReplaceWith("mapN(a, b) { (a,b) -> z }")
+  )
   fun <A, B, Z> map(
     a: Kind<F, A>,
     b: Kind<F, B>,
     lbd: (Tuple2<A, B>) -> Z
   ): Kind<F, Z> =
     a.product(b).map(lbd)
 
+  fun <A, B, Z> mapN(
+    a: Kind<F, A>,
+    b: Kind<F, B>,
+    lbd: (Tuple2<A, B>) -> Z
+  ): Kind<F, Z> =
+    a.product(b).map(lbd)
+
+  @Deprecated(
+    "map is being renamed to mapN",
+    ReplaceWith("mapN(a, b, c) { (a,b,c) -> z }")
```

Comment:
Can this refactor the lambda? I've seen mixed results with `ReplaceWith` where I see it work well and very poorly.

Reply:
No it does not, l'll fix it now!